src/library.js

```diff
@@ -3771,13 +3771,16 @@ LibraryManager.library = {
   },
 
   // Callable in pthread without __proxy needed.
+  emscripten_exit_with_live_runtime__sig: 'v',
+  emscripten_exit_with_live_runtime__deps: ['$runtimeKeepalivePush'],
   emscripten_exit_with_live_runtime: function() {
 #if !MINIMAL_RUNTIME
-    noExitRuntime = true;
+    runtimeKeepalivePush();
```

Comment:
this is sometimes ifdefed and sometimes not. is it not enough to assume closure will remove the empty function, and so the ifdefs can be removed?

Reply:
Another option is to write `{{{ makeRuntimeKeepAlivePush() }}}` and return an empty string for minimal runtime. that might be better?

Reply:
Yes, I tried doing this both way.    I found that calls to push/pop were so common (in the browser.py file etc) that ifdef'ing affected readability significantly so I decided to go for the empty function approach.   Given that most of the code that used push/pop is in `browser.py` which is generally no used in MINIMAL_RUNTIME much I think its reasonable compromise.

I'll update this to move outside the ifdef.    

Another approach (if we find that these null call do appear a lot in MINIMAL_RUNTIME output, which I don't think they will) is to use something like `{{{ runtimeKeepalivePush() }}}`

Reply:
I agree a lot of ifdefing would be bad, yeah.

If there is very little code that minimal runtime may use with this, then maybe ifdefing those few spots makes sense? Or `{{{ }}}` sounds ok. I guess either could be a followup if we want.